### DISCUSSION TITLE
[QUESTION] - Woocommerce plugin

### DISCUSSION THREAD

**Discussion by marijang** [infinum/eightshift-frontend-libs#792] [General]
Hey i want to create plugin just like eightshift forms but for support woocommerce templates.

1. I want to use block and components in plugin 
i don't know how to include them in plugin i just try wp boilerplate initi plugin but...

2. I want to use components from theme, components like heading, paragraph and blocks like columns, card....

3. i want to use reusable blocks for product page for example etc....

Can you help with that

**Comment by mbmjertan** [comment]
1. `wp boilerplate blocks` will give you a list of commands you can use
2. could be tricky to reuse components from theme in plugin and I am not sure that you want to do that architecturally? depends what you want to achieve. Eightshift Forms does not do that afaik, it just adds new blocks (and components internally) - you can use it without an Eightshift theme installed. you can use both theme and plugin blocks in gutenberg. it may be better to just implement the render logic in the theme using hooks?
3. `wp boilerplate create admin-reusable-blocks-menu` and implement logic to render your reusable blocks on product pages

**Comment by marijang** [comment]
I try with wp boilerplate block but it seems that i do not have assets folder so there is no scripts at all. I can add blocks but i cant see them into wp.

I change block category to eightshift-woocommerce

**Comment by marijang** [comment]
<img width="591" alt="Screen Shot 2023-12-05 at 10 31 56" src="https://github.com/infinum/eightshift-frontend-libs/assets/1381443/7ad69c2c-f5f1-426b-83bb-50cdd351cc9a">
